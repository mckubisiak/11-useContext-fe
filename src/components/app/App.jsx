import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import ApiContainer from '../apiContainer';

export default function App() {
  return (
    <BrowserRouter>
      <Switch>
        <Route path="/" exact component={ApiContainer} />
      </Switch>
    </BrowserRouter>
  );
}
