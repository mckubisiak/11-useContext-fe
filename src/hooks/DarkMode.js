import React, { createContext, useReducer } from 'react';

export const ModeContext = createContext();

const initialState = {
  darkMode: true,
};

const modeReducer = (state, action) => {
  switch (action.type) {
    case 'LIGHTMODE':
      return { darkMode: false };
    case 'DARKMODE':
      return { darkMode: true };
    default:
      return state;
  }
};

export function ModeProvider(props) {
  const [state, dispatch] = useReducer(modeReducer, initialState);

  return (
    <ModeContext.Provider value={{ state, dispatch }}>
      {props.children}
    </ModeContext.Provider>
  );
}
