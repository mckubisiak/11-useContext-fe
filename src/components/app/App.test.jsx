import React from 'react';
import { screen, render } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import characterlist from '../fixtures/characterlist.json';
import userEvent from '@testing-library/user-event';

import App from './App';
import { ModeProvider } from '../../hooks/DarkMode';

const mockServer = setupServer(
  rest.get('https://rickandmortyapi.com/api/character/', (req, res, ctx) => {
    return res(ctx.json(characterlist));
  })
);

describe('RickAndMortyApi', () => {
  beforeAll(() => mockServer.listen());
  afterAll(() => mockServer.close());

  it('renders lists of characters', async () => {
    render(
      <ModeProvider>
        <App />
      </ModeProvider>
    );

    screen.getByAltText('loading spinner');

    const button = await screen.getByRole('button', {
      name: 'mode',
    });
    const ul = await screen.findByRole('list', { name: 'characters' });
    const div = screen.getByTestId('div');

    
    expect(ul).toMatchSnapshot();
    
    expect(div).toHaveClass('bg bgDark');
    userEvent.click(button);
    
    expect(div).toHaveClass('bg bgLight');
    
  });
});
