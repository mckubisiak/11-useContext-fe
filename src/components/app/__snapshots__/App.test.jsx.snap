// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RickAndMortyApi renders lists of characters 1`] = `
<ul
  aria-label="characters"
  name="characters"
>
  <li>
    <p>
      1
    </p>
    <p>
      Rick Sanchez
    </p>
    <img
      alt="Rick Sanchez"
      src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
    />
  </li>
  <li>
    <p>
      2
    </p>
    <p>
      Morty Smith
    </p>
    <img
      alt="Morty Smith"
      src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
    />
  </li>
  <li>
    <p>
      3
    </p>
    <p>
      Summer Smith
    </p>
    <img
      alt="Summer Smith"
      src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
    />
  </li>
  <li>
    <p>
      4
    </p>
    <p>
      Beth Smith
    </p>
    <img
      alt="Beth Smith"
      src="https://rickandmortyapi.com/api/character/avatar/4.jpeg"
    />
  </li>
  <li>
    <p>
      5
    </p>
    <p>
      Jerry Smith
    </p>
    <img
      alt="Jerry Smith"
      src="https://rickandmortyapi.com/api/character/avatar/5.jpeg"
    />
  </li>
  <li>
    <p>
      6
    </p>
    <p>
      Abadango Cluster Princess
    </p>
    <img
      alt="Abadango Cluster Princess"
      src="https://rickandmortyapi.com/api/character/avatar/6.jpeg"
    />
  </li>
  <li>
    <p>
      7
    </p>
    <p>
      Abradolf Lincler
    </p>
    <img
      alt="Abradolf Lincler"
      src="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
    />
  </li>
  <li>
    <p>
      8
    </p>
    <p>
      Adjudicator Rick
    </p>
    <img
      alt="Adjudicator Rick"
      src="https://rickandmortyapi.com/api/character/avatar/8.jpeg"
    />
  </li>
  <li>
    <p>
      9
    </p>
    <p>
      Agency Director
    </p>
    <img
      alt="Agency Director"
      src="https://rickandmortyapi.com/api/character/avatar/9.jpeg"
    />
  </li>
  <li>
    <p>
      10
    </p>
    <p>
      Alan Rails
    </p>
    <img
      alt="Alan Rails"
      src="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
    />
  </li>
  <li>
    <p>
      11
    </p>
    <p>
      Albert Einstein
    </p>
    <img
      alt="Albert Einstein"
      src="https://rickandmortyapi.com/api/character/avatar/11.jpeg"
    />
  </li>
  <li>
    <p>
      12
    </p>
    <p>
      Alexander
    </p>
    <img
      alt="Alexander"
      src="https://rickandmortyapi.com/api/character/avatar/12.jpeg"
    />
  </li>
  <li>
    <p>
      13
    </p>
    <p>
      Alien Googah
    </p>
    <img
      alt="Alien Googah"
      src="https://rickandmortyapi.com/api/character/avatar/13.jpeg"
    />
  </li>
  <li>
    <p>
      14
    </p>
    <p>
      Alien Morty
    </p>
    <img
      alt="Alien Morty"
      src="https://rickandmortyapi.com/api/character/avatar/14.jpeg"
    />
  </li>
  <li>
    <p>
      15
    </p>
    <p>
      Alien Rick
    </p>
    <img
      alt="Alien Rick"
      src="https://rickandmortyapi.com/api/character/avatar/15.jpeg"
    />
  </li>
  <li>
    <p>
      16
    </p>
    <p>
      Amish Cyborg
    </p>
    <img
      alt="Amish Cyborg"
      src="https://rickandmortyapi.com/api/character/avatar/16.jpeg"
    />
  </li>
  <li>
    <p>
      17
    </p>
    <p>
      Annie
    </p>
    <img
      alt="Annie"
      src="https://rickandmortyapi.com/api/character/avatar/17.jpeg"
    />
  </li>
  <li>
    <p>
      18
    </p>
    <p>
      Antenna Morty
    </p>
    <img
      alt="Antenna Morty"
      src="https://rickandmortyapi.com/api/character/avatar/18.jpeg"
    />
  </li>
  <li>
    <p>
      19
    </p>
    <p>
      Antenna Rick
    </p>
    <img
      alt="Antenna Rick"
      src="https://rickandmortyapi.com/api/character/avatar/19.jpeg"
    />
  </li>
  <li>
    <p>
      20
    </p>
    <p>
      Ants in my Eyes Johnson
    </p>
    <img
      alt="Ants in my Eyes Johnson"
      src="https://rickandmortyapi.com/api/character/avatar/20.jpeg"
    />
  </li>
</ul>
`;
